src/plugins/navigation_embeddable/common/persistable_state/references.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Reference } from '@kbn/content-management-utils';
+import { DASHBOARD_LINK_TYPE, NavigationEmbeddableAttributes } from '../content_management';
+
+export function extractReferences({
+  attributes,
+  references = [],
+}: {
+  attributes: NavigationEmbeddableAttributes;
+  references?: Reference[];
+}) {
+  if (!attributes.links) {
+    return { attributes, references };
+  }
+
+  const { links } = attributes;
+  const extractedReferences: Reference[] = [];
+  links.forEach((link) => {
+    if (link.type === DASHBOARD_LINK_TYPE) {
+      const refName = `link_${link.id}_dashboard`;
+      extractedReferences.push({
+        name: refName,
+        type: 'dashboard',
+        id: link.destination,
+      });
+      link.destination = refName;
```

Comment:
Not sure if this is the best practice. [Maps](https://github.com/elastic/kibana/blob/main/x-pack/plugins/maps/common/migrations/references.ts#L95-L96) and [Dashboard](https://github.com/elastic/kibana/blob/main/src/plugins/dashboard/common/dashboard_container/persistable_state/dashboard_container_references.ts#L124-L132) add a different property for the `refName` rather than overwriting the existing property. But they also serialize the attributes into a JSON string for saved objects. Links saves the attributes as JSON, so adding another property (e.g. `refName`) to the `NavigationEmbeddableAttributes` seems unwarranted. Unless I'm missing something? cc @ThomThomson

Reply:
Personally, the swip-swap of `destination` referring to `refName` made tracing the tests hurt my brain a little 😆 Maybe it's just me? My preference is usually clarity in cases like this (i.e. just keeping `destination` as always the dashboard ID, and having a separate `refName` property) - but I don't necessarily feel **super** strong about this.

Reply:
Using a `refname` property on dashboard links would likely require separate types for dashboard and external links since external links won't have references. I can spend a bit more time digging in if the current method is concerning. But I'll probably focus on getting https://github.com/elastic/kibana/issues/164322 fixed first as it may simplify some bits.